besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -2233,6 +2238,31 @@ public EnodeDnsConfiguration getEnodeDnsConfiguration() {
     return enodeDnsConfiguration;
   }
 
+  private void checkPortClash() {
+    // List of port parameters
+    final List<Integer> ports =
+        asList(
+            p2pPort,
+            graphQLHttpPort,
+            rpcHttpPort,
+            rpcWsPort,
+            metricsPort,
+            metricsPushPort,
+            stratumPort);
+    ports.stream()
+        .filter(Objects::nonNull)
+        .forEach(
+            port -> {
+              if (!allocatedPorts.add(port)) {
```

Comment:
Looks like we need to special case port zero, which is designed to conflict.
```suggestion
              if (port != 0 && !allocatedPorts.add(port)) {
```
We will also need a unit test for this.